lua/workspace-diagnostics/init.lua

```diff
@@ -110,7 +115,7 @@ function M.populate_workspace_diagnostics(client, bufnr)
       goto continue
     end
 
-    if not vim.tbl_contains(client.config.filetypes, filetype) then
+    if not (vim.tbl_contains(client.config.filetypes, filetype) or vim.tbl_contains(M.options.exclude_filetypes, filetype)) then
```

Comment:
I don't understand how any of those excluded filetypes can satisfy the original if? That would mean that there is an lsp client that supports those filetypes. Could you please clarify?

Reply:
Yeah, looking into it some more, it looks like this may be a work-around to a tsserver problem I was getting where when I opened my Nextjs project, it would open my mp4 files. I am uncertain as to why, as I have not done any custom configuration of tsserver. Checking my tsserver config, I see that
```
['config'] = {
	['filetypes'] = {
		[1] = 'javascript',
		[2] = 'javascriptreact',
		[3] = 'javascript.jsx',
		[4] = 'typescript',
		[5] = 'typescriptreact',
		[6] = 'typescript.tsx'
	},
```
I also have jpeg, svp, wav, and mp3 files, and none of them have opened, so I think this must be a problem on my side. I will eliminate this and solve the issue on my side.

Reply:
I have made a pull request with the changes